src/spikeinterface/postprocessing/tests/test_template_metrics.py

```diff
@@ -3,6 +3,36 @@
 import pytest
 
 
+def test_compute_new_template_metrics(small_sorting_analyzer):
+    """
+    Computes template metrics then computes a subset of quality metrics, and checks
+    that the old quality metrics are not deleted.
+
+    Then computes template metrics with new parameters and checks that old metrics
+    are deleted.
+    """
+
+    small_sorting_analyzer.compute("template_metrics")
+    small_sorting_analyzer.compute({"template_metrics": {"metric_names": ["exp_decay"]}})
+
+    template_metric_extension = small_sorting_analyzer.get_extension("template_metrics")
+
+    # Check old metrics are not deleted and the new one is added to the data and metadata
+    assert "exp_decay" in list(template_metric_extension.get_data().keys())
+    assert "half_width" in list(template_metric_extension.get_data().keys())
+
+    # check that, when parameters are changed, the old metrics are deleted
+    small_sorting_analyzer.compute(
+        {"template_metrics": {"metric_names": ["exp_decay"], "metrics_kwargs": {"recovery_window_ms": 0.6}}}
+    )
+
+    template_metric_extension = small_sorting_analyzer.get_extension("template_metrics")
+
+    assert "half_width" not in list(template_metric_extension.get_data().keys())
```

Comment:
same here might as well check against full default list if it was indeed all computed in the first call